app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuSitePermissionsExceptionsRobot.kt

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@file:Suppress("TooManyFunctions")
+
+package org.mozilla.fenix.ui.robots
+
+/**
+ * Implementation of Robot Pattern for the settings Site Permissions Notification sub menu.
+ */
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.Visibility
+import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import org.hamcrest.CoreMatchers.allOf
+import org.mozilla.fenix.helpers.click
+
+class SettingsSubMenuSitePermissionsExceptionsRobot {
+
+    fun verifyNavigationToolBarHeader() = assertNavigationToolBarHeader()
+
+    fun verifyExceptionDefault() = assertExceptionDefault()!!
+
+    fun verifySitePermissionsExceptionSubMenuItems() {
+        verifyExceptionDefault()
+    }
+
+    class Transition {
+        val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())!!
+
+        fun goBack(interact: SettingsSubMenuSitePermissionsRobot.() -> Unit): SettingsSubMenuSitePermissionsRobot.Transition {
+            mDevice.waitForIdle()
+            goBackButton().click()
+
+            SettingsSubMenuSitePermissionsRobot().interact()
+            return SettingsSubMenuSitePermissionsRobot.Transition()
+        }
+    }
+}
+
+private fun goBackButton() =

```

Comment:
Wondering if we could add this to one of the helpers files...this is defined the same way in many different robots... @AaronMT wdyt?